google/auth/crypt.py

```diff
@@ -227,3 +227,18 @@ def from_string(cls, key, key_id=None):
             raise ValueError('No key could be detected.')
 
         return cls(private_key, key_id=key_id)
+
+    @classmethod
+    def from_service_account_file(cls, filename):
+        """Creates a Signer instance from a service account .json file
+        in Google format.
+
+        Args:
+            filename (str): The path to the service account .json file.
+
+        Returns:
+            Signer: The constructed signer.
+        """
+        from google.auth import _service_account_info
```

Comment:
Because right now `_service_account_info` does a really good job of containing all of the logic needed to parse and validate service account files. I'd need to either move or duplicate some of that logic into crypt (breaking single-responsibility). [This logic in particular](https://github.com/GoogleCloudPlatform/google-auth-library-python/blob/master/google/auth/_service_account_info.py#L45) which constitues the majority of the logic in that module.